### ISSUE TITLE
Display remarks in buffer as overlays?

### ISSUE THREAD

**Issue by ryanprior** [nobiot/org-remark#13]
Hi there! This looks fantastic, I love all the thoughtful touches you've put on it.

For some time I've wanted an alternative to [annotate.el](https://github.com/bastibe/annotate.el) that stores marginalia using org-mode instead of a big alist. I love the feature of annotate.el that it can show the text as an overlay directly in the buffer and would miss it if I switched over now–would you be interested in adding that capability to org-remark?

**Comment by nobiot** [comment]
Hi @ryanprior 
Thank you.

I think showing the text as an overlay from an Org file would be much beyond my capability. See this [exchange](https://github.com/bastibe/annotate.el/issues/91) in annote.el's repository. You could consider something like using [postframe](https://github.com/randomwangran/org-marginalia-posframe) that @randomwangran has done.

The exchange was from rather long ago so the context may have changed, and my memory has faded; I think annote.el and org-remark serve different purposes. The former is probably for a very short commentary. I believe its main application is for source code. My motivation for Org-remark comes from writing / note-taking, and I wanted to have means to record comments (often rather long) alongside the main text (but separate).

**Comment by ryanprior** [comment]
Thank you for these links! I think I will continue to use annotate.el for my source code (perhaps I can modify it to make use of an org-file as a db) and try to introduce org-remark for longer notes.

It seems that for first class general-purpose transclusion of text with different modes, we may need some editor-level support as overlays are limited.

**Comment by randomwangran** [comment]
Sorry for the extremely long delay in response to many comments from @nobiot. Especially, for the issues in your `org-transclusion` repository. My attention shifted to something else last year, but I do use `org-transclusion` on daily basis.

Regarding this issue, I have not used the package (`org-remark`) since I've done that extension. I did not read [the source code](https://github.com/OpenFOAM/OpenFOAM-dev) as I used to.  I would like to continue exploring the code this year and hope to make a consistent habit in 2022, so I would make improvements based on `org-remark` and `org-roam`, especially for the preview function.

Ran
